sdk/openai/openai-assistants/README.md

```diff
@@ -0,0 +1,347 @@
+
+**Note:** Sometimes the API key is referred to as a "subscription key" or "subscription API key."
+
+```PowerShell
+az cognitiveservices account keys list --resource-group <your-resource-group-name> --name <your-resource-name>
+```
+
+## Key concepts
+
+See [OpenAI's "how assistants work"](https://platform.openai.com/docs/assistants/how-it-works) documentation for an
+overview of the concepts and relationships used with assistants. This overview closely follows
+[OpenAI's overview example](https://platform.openai.com/docs/assistants/overview) to demonstrate the basics of
+creating, running, and using assistants and threads.
+
+To get started, create an `AssistantsClient`:
+```javascript Snippet:OverviewCreateClient
+const assistantsClient = new AssistantsClient("<endpoint>", new AzureKeyCredential("<azure_api_key>"));
+```
+
+With a client, an assistant can then be created. An assistant is a purpose-built interface to OpenAI models that can call Tools while allowing high-level instructions throughout the lifetime of the assistant.
+
+The code to create an assistant:
+```javascript Snippet:OverviewCreateAssistant
+const assistant = await assistantsClient.assistants.createAssistant({
+  model: "gpt-4-1106-preview",
+  name: "JS Math Tutor",
+  instructions: "You are a personal math tutor. Write and run code to answer math questions.",
+  tools: [{ type: "code_interpreter" }]
+});
+```
+
+A conversation session between an Assistant and a user is called a Thread. Threads store Messages and automatically handle truncation to fit content into a model’s context.
+
+To create a thread:
+```javascript Snippet:OverviewCreateThread
+const assistantThread = await assistantsClient.assistantThreads.createThread();
+```
+
+Message represent a message created by an Assistant or a user. Messages can include text, images, and other files. Messages are stored as a list on the Thread.
+With a thread created, messages can be created on it:
+```javascript Snippet:OverviewCreateMessage
+const question = "I need to solve the equation '3x + 11 = 14'. Can you help me?";
+const messageResponse = await assistantsClient.threadMessages.createMessage(assistantThread.id, "user", question);
+```
+
+A Run represent an invocation of an Assistant on a Thread. The Assistant uses it’s configuration and the Thread’s Messages to perform tasks by calling models and tools. As part of a Run, the Assistant appends Messages to the Thread.
+A run can then be started that evaluates the thread against an assistant:
+```javascript Snippet:OverviewCreateRun
+let runResponse = await assistantsClient.threadRuns.createRun(assistantThread.id, assistant.id, {
+   instructions: "Please address the user as Jane Doe. The user has a premium account." 
+});
+```
+
+Once the run has started, it should then be polled until it reaches a terminal status:
+```javascript Snippet:OverviewWaitForRun
+do {
+  await new Promise((resolve) => setTimeout(resolve, 500));
```

Comment:
Do we know how long runs take in the typical case? I imagine this will be a function in the number of messages on the thread and how large their content are but I am mainly wonder if polling every 0.5 second is a bit aggressive?

Reply:
Perhaps, changed to 800ms